examples/js/exporters/GLTFExporter.js

```diff
@@ -459,21 +473,36 @@ THREE.GLTFExporter.prototype = {
 			var mimeType = map.format === THREE.RGBAFormat ? 'image/png' : 'image/jpeg';
 			var gltfImage = { mimeType: mimeType };
 
-			if ( options.embedImages ) {
+			if ( options.embedImages || ( options.forcePowerOfTwoTexture && ! isPowerOfTwo( map.image ) ) ) {
 
 				var canvas = cachedCanvas = cachedCanvas || document.createElement( 'canvas' );
+
 				canvas.width = map.image.width;
 				canvas.height = map.image.height;
+
+				if ( options.forcePowerOfTwoTexture ) {
```

Comment:
Wouldn't it make more sense to move the previous check here?

```js
if ( options.forcePowerOfTwoTexture && ! isPowerOfTwo( map.image ) ) {
```

So we only do the resize when `embedImages` is `true` , and if we do the resize we show the warning...